packages/perseus/src/widgets/radio/choice-none-above.jsx

```diff
@@ -1,54 +1,60 @@
-class ChoiceNoneAbove extends React.Component<$FlowFixMe> {
-    _choice: $FlowFixMe;
-    static propTypes = {
-        className: PropTypes.string,
-        content: PropTypes.node,
-        showContent: PropTypes.bool,
+import type {ChoiceProps} from "./choice.jsx";
+
+type Props = {|
+    ...ChoiceProps,
+    showContent: boolean,
+|};
+
+type WithForwardRef = {|forwardedRef: React.Ref<"button">|};
+
+type PropsWithForwardRef = {|
+    ...Props,
+    ...WithForwardRef,
+|};
+
+function ChoiceNoneAbove(props: PropsWithForwardRef): React.Node {
+    const {showContent, content, forwardedRef, ...rest} = props;
+
+    const choiceProps = {
+        ...rest,
+        content: showContent ? (
+            content
+        ) : (
+            // We use a Renderer here because that is how
+            // `this.props.content` is wrapped otherwise.
+            // We pass in a key here so that we avoid a semi-spurious
+            // react warning when we render this in the same place
+            // as the previous choice content renderer.
+            // Note this destroys state, but since all we're doing
+            // is outputting "None of the above", that is okay.
+            <Renderer
+                key="noneOfTheAboveRenderer"
+                content={i18n._("None of the above")}
+            />
+        ),
     };
 
-    static defaultProps: $FlowFixMe = {
-        showContent: true,
-    };
+    return <Choice {...choiceProps} ref={forwardedRef} />;
+}
 
-    focusInput: () => void = () => {
-        this._choice.focusInput();
-    };
+ChoiceNoneAbove.defaultProps = {
+    showContent: true,
+};
 
-    render(): React.Node {
-        const choiceProps = {
-            ...this.props,
-            className: classNames(this.props.className, "none-of-above"),
-            content: this.props.showContent ? (
-                this.props.content
-            ) : (
-                // We use a Renderer here because that is how
-                // `this.props.content` is wrapped otherwise.
-                // We pass in a key here so that we avoid a semi-spurious
-                // react warning when we render this in the same place
-                // as the previous choice content renderer.
-                // Note this destroys state, but since all we're doing
-                // is outputting "None of the above", that is okay.
-                <Renderer
-                    key="noneOfTheAboveRenderer"
-                    content={i18n._("None of the above")}
-                />
-            ),
-            ref: (el) => (this._choice = el),
-        };
-
-        return <Choice {...choiceProps} />;
-    }
-}
+type ExportProps = $Diff<
+    React.ElementConfig<typeof ChoiceNoneAbove>,
+    WithForwardRef,
+>;
 
-export default ChoiceNoneAbove;
+export default (React.forwardRef<ExportProps, HTMLButtonElement>(
+    (props, ref) => <ChoiceNoneAbove {...props} forwardedRef={ref} />,
+): React.AbstractComponent<ExportProps, HTMLButtonElement>);
```

Comment:
This is all just for forwarding a ref

Reply:
Flow is just the worst when it comes to export types. 😞